### PR TITLE
Update `@yamada-ui/table` accessibility

### LINKED ISSUE #1635: Update `@yamada-ui/table` accessibility
### Description

his issue aims to address accessibility deficiencies in our table component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

WAI-ARIA Roles, States, and Properties
- [ ] If there is an element in the user interface that serves as a label for the table, [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) is set on the table element with a value that refers to the labelling element. Otherwise, a label is specified for the table element using [aria-label](https://w3c.github.io/aria/#aria-label).
- [ ] If the table has a caption or description, [aria-describedby](https://w3c.github.io/aria/#aria-describedby) is set on the table element with a value referring to the element containing the description.
- [x] If the table contains sortable columns or rows, [aria-sort](https://w3c.github.io/aria/#aria-sort) is set to an appropriate value on the header cell element for the sorted column or row as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/#gridAndTableProperties_sort).
- [x] If there are conditions where some rows or columns are hidden or not present in the DOM, e.g., there are widgets on the page for hiding rows or columns, the following properties are applied as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/).
  - [x] [aria-colcount](https://w3c.github.io/aria/#aria-colcount) or [aria-rowcount](https://w3c.github.io/aria/#aria-rowcount) is set to the total number of columns or rows, respectively.
  - [x] [aria-colindex](https://w3c.github.io/aria/#aria-colindex) or [aria-rowindex](https://w3c.github.io/aria/#aria-rowindex) is set to the position of a cell within a row or column, respectively.
- [ ] If the table includes cells that span multiple rows or multiple columns, then [aria-rowspan](https://w3c.github.io/aria/#aria-rowspan) or [aria-colspan](https://w3c.github.io/aria/#aria-colspan) is applied as described in the [Grid and Table Properties Practice](https://www.w3.org/WAI/ARIA/apg/practices/grid-and-table-properties/#gridAndTableProperties_spans).

###  Problem Statement/Justification
none

###  Proposed Solution or API
No response

###  Alternatives
none 

### Are you willing to participate in implementing this feature and create a pull request include the implementation?
None

###  Additional Information
No response

### PR DESCRIPTION
**Pull request by cidkumagai**
Closes #1635 

## Description

Update @yamada-ui/table accessibility

## Additional Information

The current implementation of the component does not support the following ARIA attributes:

	1.	aria-labelledby: This attribute should be set on the table element with a value referring to an element that serves as a label for the table. If no such element exists, aria-label should be used to specify a label for the table element.
	2.	aria-describedby: This attribute should be set on the table element with a value referring to an element containing a caption or description of the table.
	3.	aria-rowspan and aria-colspan: These attributes should be applied to cells that span multiple rows or columns, respectively, as described in the Grid and Table Properties Practice.

Currently, these ARIA attributes do not seem to be included in the component, and handling for these specific requirements appears to be unimplemented.

### CODE REVIEW

**Review by 108yen — APPROVED** [review]

File: packages/components/table/src/thead.tsx [hunk 1]
```diff
@@ -84,8 +95,11 @@ export const Thead = ({ ...rest }: TableHeadProps) => {
               const resolvedColSpan = (customColSpan ?? colSpan) || 1
               const resolvedRowSpan = (customRowSpan ?? rowSpan) || 1
 
+              const isSorted = getIsSorted()
+
               const props = {
                 "aria-label": ariaLabel,
+                "aria-sort": !isSorted ? ariaSort.none : ariaSort[isSorted],
```
Comment (suzukisan22) [inline packages/components/table/src/thead.tsx:102]: nits
```suggestion
                "aria-sort": isSorted ? ariaSort[isSorted] : ariaSort.none,
```

**Review by 108yen — APPROVED** [review]

**Review by suzukisan22 — APPROVED** [review]
LGTM！

File: packages/components/table/src/thead.tsx [hunk 2]
```diff
@@ -15,8 +15,19 @@ import type { SortDirection, Column } from "./use-table"
 import { useTableContext, render } from "./use-table"
 
 export type TableHeadProps = NativeTableHeadProps
+export type AriaSort = {
```
Comment (hirotomoyamada) [inline packages/components/table/src/thead.tsx:18]: Please delete `AriaSort`.

File: packages/components/table/src/thead.tsx [hunk 3]
```diff
@@ -15,8 +15,19 @@ import type { SortDirection, Column } from "./use-table"
...
+  none: "none"
+  asc: "ascending"
+  desc: "descending"
+}
 
 export const Thead = ({ ...rest }: TableHeadProps) => {
+  const ariaSort: AriaSort = {
```
Comment (hirotomoyamada) [inline packages/components/table/src/thead.tsx:25]: Please delete `ariaSort`.

It is also not necessary to define it within the function component. If you do, define it outside the function component.

File: packages/components/table/src/thead.tsx [hunk 4]
```diff
@@ -84,8 +95,11 @@ export const Thead = ({ ...rest }: TableHeadProps) => {
               const resolvedColSpan = (customColSpan ?? colSpan) || 1
               const resolvedRowSpan = (customRowSpan ?? rowSpan) || 1
 
+              const isSorted = getIsSorted()
+
               const props = {
                 "aria-label": ariaLabel,
+                "aria-sort": isSorted ? ariaSort[isSorted] : ariaSort.none,
```
Comment (hirotomoyamada) [inline packages/components/table/src/thead.tsx:102]: The `"aria-sort"` line should complete it all.

```
"aria-sort": isSorted ? isSorted === "asc" ? "ascending" : "descending" : "none"
```

Comment (cidkumagai) [reply packages/components/table/src/thead.tsx:102]: ` 型 'string' を型 '"none" | "ascending" | "descending" | "other" | undefined' に割り当てることはできません。`
It seems that type definitions might be necessary to address this error. What do you think?

Comment (hirotomoyamada) [reply packages/components/table/src/thead.tsx:102]:
```
const props: ThProps = {
```

Comment (cidkumagai) [reply packages/components/table/src/thead.tsx]: Thank you very much!
I was able to fix it as advised, so please re-review!

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**